### PR TITLE
CLOUDSTACK-9104: VM naming convention in case vmware is used

### PR DESCRIPTION
**Pull request by priyankparihar**
I have reverted all the changes. Now functionality is same as it was  in earlier version. 
ACS Link ==>
https://issues.apache.org/jira/browse/CLOUDSTACK-9104

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@priyankparihar I do not know any vmware users
@bhaisaab Do you have any insight on this change? or do you know any candidate reviewer for this?

File: plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java [hunk 1]
```diff
@@ -1889,11 +1889,20 @@ int getReservedCpuMHZ(VirtualMachineTO vmSpec) {
...
+
+        String vmInternalCSName = null;
+        String vmNameOnVcenter  = null;
+        if(vmSpec != null)
+        {
+            vmInternalCSName = vmNameOnVcenter = vmSpec.getName();
+            if (_instanceNameFlag == true) {
+                String[] tokens = vmInternalCSName.split("-");
+                assert (tokens.length >= 3); // vmInternalCSName has format i-x-y-<instance.name>
+                vmNameOnVcenter = String.format("%s-%s-%s-%s", tokens[0], tokens[1], tokens[2], vmSpec.getHostName());
```
Comment (bhaisaab) [inline plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java]: @priyankparihar can you comment on the issues and motivation around this? Do you think we have users who would want to keep hostname as VM names on vCenter side if the global setting is enabled (the _instanceNameFlag is true).

If this is not true, by default the name (as returned by the vmSpec) is like: i-[0-9]*-[0-9]*-VM. (The suffix "VM", is taken from another global setting where the default value is "VM").

**Comment by bhaisaab** [comment]
@DaanHoogland yes I can do that, I'm personally dealt with a lot of vmware/cloudstack naming issues. The syntax/pattern has changes several times in the past between 4.0, 4.2, 4.3 causing VM sync issues when users upgrade.

**Comment by DaanHoogland** [comment]
@bhaisaab @priyankparihar what is the verdict on this change?

**Comment by priyankparihar** [comment]
@bhaisaab 
User with different account cannot create VMs with the same name, which was possible earlier (I am not sure in which CCP version). That time naming convention used was like this “I-<user-id>-<Display-Name>”
Currently if vm.instancename.flag is set to true the VM name will be exactly as display name given.

File: plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java (hunk 1 above)
Comment (bhaisaab) [inline plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java]: @priyankparihar alright, even if we go with this change. If vmSpec.getHostName() was null then we've a problem with the vm name (if _instanceNameFlag, i.e. the global setting was true).

Once you put a null check, it should be good to go.

**Comment by bhaisaab** [comment]
@priyankparihar Fair enough, please see a code review comment. Once you fix the null check, I think your PR is good to go. Also, write a unit test to enforce the changes; i.e. the unit test should break in future if someone tries to change the VM naming convention in future.

File: plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java (hunk 1 above)
Comment (bhaisaab) [inline plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java]: Do something like;
String hostString = vmSpec.getHostName();
if (hostString == null) {
  hostString = // get value from the vm suffix global setting (by default "VM");
}

**Comment by priyankparihar** [comment]
@bhaisaab I think null check is not required. I have verified( because in case of null if automatically picks  some random value). But if you still think it may break something then please notify.

**Comment by bhaisaab** [comment]
@priyankparihar I won't be able to give a LGTM without the null check, split string related code and a unit test that breaks if someone tries to change the convention in future

**Pull request closed without merging**